### PR TITLE
Task 3

### PR DESCRIPTION
**Pull request by DavydovDenys**

### CODE REVIEW

File: lib/transport.rb [hunk 1]
```diff
@@ -15,6 +17,43 @@ def initialize(max_weight, speed, available)
     @max_weight = max_weight
     @speed = speed.to_f
     @available = available
+    @location = 'in garage'
+    @number_of_deliveries = 0
+    @delivery_cost = 0
+  end
+
+  def self.filter_by(attribute, &block)
+    objects_with_given_attributes = @@park.select { |p| p.respond_to?(attribute) }
```
Comment (dmgrinko1) [inline lib/transport.rb]: you can use here park.select cause you already define it

File: lib/transport.rb [hunk 2]
```diff
@@ -15,6 +17,43 @@ def initialize(max_weight, speed, available)
...
+    return objects_with_given_attributes.to_enum(:each) unless block_given?
+
+    result = []
+    objects_with_given_attributes.each do |object|
+      if block.call(object.send(attribute))
```
Comment (dmgrinko1) [inline lib/transport.rb]: better to use here public_send - read documentation about it please

File: lib/transport.rb [hunk 3]
```diff
@@ -15,6 +17,43 @@ def initialize(max_weight, speed, available)
...
+    end
+    result
+  end
+
+  def self.find_by(attribute, value)
+    objects_with_given_attributes = @@park.select { |p| p.respond_to?(attribute) }
+
+    result = nil
+    objects_with_given_attributes.each do |object|
+      if object.send(attribute) == value
```
Comment (dmgrinko1) [inline lib/transport.rb]: better to use here public_send - read documentation about it please

File: lib/transport.rb [hunk 4]
```diff
@@ -15,6 +17,43 @@ def initialize(max_weight, speed, available)
...
+    return objects_with_given_attributes.to_enum(:each) unless block_given?
+
+    result = []
+    objects_with_given_attributes.each do |object|
+      if block.call(object.send(attribute))
+        result << object
+      end
+    end
+    result
+  end
```
Comment (dmgrinko1) [inline lib/transport.rb]: you can do something like this with  'map', to simplify expression 
objects_with_given_attributes.map {|t| t if block.call(object.public_send(attribute))}

File: lib/transport.rb [hunk 5]
```diff
@@ -15,6 +17,43 @@ def initialize(max_weight, speed, available)
...
+
+  def self.all
+    @@park.select { |p| p.is_a?(self) }
+  end
+
+  def park
+    @@park
```
Comment (dmgrinko1) [inline lib/transport.rb]: couple things here

def initialize - is always first method. Does not matter thich type of methods do you have
(def self.foo end or def foo end)
if you have more then 2 self methods use 'block syntax for them'
https://stackoverflow.com/questions/2505067/class-self-idiom-in-ruby
this block goes after initialize method, and all method inside without self, this construction already provide it.

File: lib/transport.rb [hunk 6]
```diff
@@ -15,6 +17,43 @@ def initialize(max_weight, speed, available)
...
+      end
+    end
+    result
+  end
+
+  def self.all
+    @@park.select { |p| p.is_a?(self) }
+  end
+
+  def park
```
Comment (dmgrinko1) [inline lib/transport.rb]: If Im not mistaken we should do something is delivery service with new attributes
@location = LOCATION[:in_garage]
    @number_of_deliveries = 0
    @delivery_cost = 0

File: lib/transport.rb (hunk 2 above)
Comment (DavydovDenys) [reply lib/transport.rb]: done

File: lib/transport.rb (hunk 3 above)
Comment (DavydovDenys) [reply lib/transport.rb]: done

File: lib/transport.rb (hunk 1 above)
Comment (DavydovDenys) [reply lib/transport.rb]: done

File: lib/transport.rb (hunk 4 above)
Comment (DavydovDenys) [reply lib/transport.rb]: done

File: lib/transport.rb (hunk 5 above)
Comment (DavydovDenys) [reply lib/transport.rb]: done

File: lib/transport.rb (hunk 6 above)
Comment (DavydovDenys) [reply lib/transport.rb]: done

**Review by dmgrinko1 — APPROVED** [review]

**Comment by dmgrinko1** [comment]
@DavydovDenys You can merge it now

**Pull request merged**